src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(
+  items: Item[],
+  key: keyof Item,
+): GroupsMap<Item> {
+  const groupedObject: GroupsMap<Item> = {};
+
+  for (const item of items) {
+    const groupKey = String(item[key]);
```

Comment:
Directly casting item[key] to a string might lead to unexpected behavior if the value is an object with a custom toString method. It's safer to ensure the property is actually a string before using it as a key.